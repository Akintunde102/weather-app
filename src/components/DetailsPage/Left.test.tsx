import React from 'react';
import { render, screen } from '@testing-library/react';
import LeftDetails from './Left';
import { MOCK_WEATHER_DETAILS } from '@/test/__mocks__/weather-details';

jest.mock('@/components/FavCityButton/FavCityButton',
    () => function FavCityMock() {
        return <div data-testid="fav-city-button"></div>
    }
);

describe('LeftDetails component', () => {

    test('renders weather details correctly', () => {

        render(
            <LeftDetails weatherDetails={MOCK_WEATHER_DETAILS} />
        );

        expect(screen.getByText('Queens, New York, USA')).toBeInTheDocument();
        expect(screen.getByText('25')).toBeInTheDocument();
        expect(screen.getByText('Feels like')).toBeInTheDocument();
        expect(screen.getByText('Precipitation is currently at 0')).toBeInTheDocument();
        expect(screen.getByText('Pressure: 1015 atm')).toBeInTheDocument();
        expect(screen.getByAltText('location icon')).toBeInTheDocument();
        expect(screen.getByTestId('fav-city-button')).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        render(
            <LeftDetails weatherDetails={MOCK_WEATHER_DETAILS} />
        );
        expect(screen.getByTestId("left-details")).toMatchSnapshot();
    });
});
