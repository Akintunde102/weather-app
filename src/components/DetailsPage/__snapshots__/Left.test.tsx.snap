// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftDetails component matches snapshot 1`] = `
<div
  class="leftContainer"
  data-testid="left-details"
>
  <div
    class="headerPill"
  >
    <div
      class="headerPillBody"
    >
      <div
        class="locationContainer"
      >
        <img
          alt="location icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          class="locationTag"
        >
          Queens, New York, USA
        </span>
      </div>
      <span
        class="rightTag"
        title="Make Favourite"
      >
        <div
          data-testid="fav-city-button"
        />
      </span>
    </div>
  </div>
  <div
    class="fullContentBar"
  >
    <div
      class="centerData"
    >
      <div
        class="tempData"
      >
        <div
          class="degreeTemp"
        >
          25
          <sup>
            °
          </sup>
        </div>
        <div
          class="wetness"
        >
          Sunny
        </div>
        <div
          class="tempStatement"
        />
      </div>
    </div>
    <div
      class="leftWeatherCardsContainer"
    >
      <div
        class="weatherCard"
        data-testid="weather-card"
        style="background: rgba(45, 47, 56, 0.7);"
      >
        <div
          class="header"
        >
          <div
            class="container"
          >
            <div
              class="content"
            >
              <div
                class="left"
              >
                <img
                  alt="icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <span>
                  Feels like
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="body"
        >
          <div
            class="dataItem"
          >
            <span>
              24
              <sup>
                °
              </sup>
            </span>
          </div>
          <div
            class="dataItemDetails"
          >
            <span>
              Precipitation is currently at 
              0
            </span>
          </div>
        </div>
        <div
          class="footerItemDetails"
        >
          <span>
            Pressure: 
            1015
             atm
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
